apps/angular/src/app/features/anime/components/anime-table/anime-table.component.ts

```diff
@@ -0,0 +1,139 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { PageEvent } from '@angular/material/paginator';
+import { Anime } from '@js-camp/core/models/anime';
+
+import { BehaviorSubject, Observable, switchMap, combineLatest, map, tap, debounceTime } from 'rxjs';
+
+import { Sort } from '@angular/material/sort';
+
+import { MatSelectChange } from '@angular/material/select';
+
+import { AnimeType } from '@js-camp/core/utils/enums/table';
+
+import { PaginationParams } from '../../../../../core/models/pagination-params';
+
+import { AnimeService } from '../../../../../core/services/anime.service';
+
+import { SearchParamsService } from './../../../../../core/services/search-params.service';
+
+const DEBOUNCE_TIME = 500;
+const PAGE_SIZE_DEFAULT = 5;
+const INITIAL_PAGE = 0;
+const DEFAULT_ANIME_COUNT = 0;
+const ORDERING_DEFAULT = 'title_eng';
+const DEFAULT_SEARCH_VALUE = '';
+
+/**
+ * Anime table component.
+ */
+@Component({
+  selector: 'camp-anime-table',
+  templateUrl: './anime-table.component.html',
+  styleUrls: ['./anime-table.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AnimeTableComponent {
+
+  /** Anime list. */
+  public readonly animeList$: Observable<readonly Anime[]>;
+
+  /** Displayed columns. */
+  public readonly displayedColumns = ['imageSrc', 'titleEnglish', 'titleJapanese', 'type', 'status', 'airedStart'] as const;
+
+  /** Number of records per page. */
+  public readonly pageSize$ = new BehaviorSubject(PAGE_SIZE_DEFAULT);
+
+  /** Anime Search field value. */
+  public searchValue = DEFAULT_SEARCH_VALUE;
+
+  /** Count of anime taken from the request. */
+  public readonly animeCount$ = new BehaviorSubject(DEFAULT_ANIME_COUNT);
+
+  /** Index of the current page.  */
+  public readonly pageIndex$ = new BehaviorSubject(INITIAL_PAGE);
+
+  /** Sorting anime on the page. */
+  public readonly pageSort$ = new BehaviorSubject(ORDERING_DEFAULT);
+
+  /** Filtering by type on the page. */
+  public readonly pageType$ = new BehaviorSubject([] as AnimeType[]);
+
+  /** Filtering by anime name. */
+  private readonly pageSearch$ = new BehaviorSubject(DEFAULT_SEARCH_VALUE);
+
+  /** Anime types. */
+  public readonly animeTypes = Object.values(AnimeType).filter(element => typeof element === 'string');
+
+  public constructor(
+    private readonly animeService: AnimeService,
+    private readonly searchParamsService: SearchParamsService,
+  ) {
```

Comment:
If you don't use services anywhere except the constructor, you could remove `private readonly`. This way services would be treated as function arguments and won't be declared as class properties
```suggestion
  public constructor(
    animeService: AnimeService,
    searchParamsService: SearchParamsService,
  ) {
```